react/features/base/react/components/native/SlidingView.js

```diff
@@ -0,0 +1,245 @@
+// @flow
+
+import React, { PureComponent, type Node } from 'react';
+import {
+    Animated,
+    Dimensions,
+    TouchableWithoutFeedback,
+    View
+} from 'react-native';
+
+import { type StyleType } from '../../../styles';
+
+import styles from './slidingviewstyles';
+
+/**
+ * The type of the React {@code Component} props of {@link SideBar}.
+ */
+type Props = {
+
+    /**
+     * The children of {@code SideBar}.
+     */
+    children: Node,
+
+    /**
+     * Callback to notify the containing {@code Component} that the view is
+     * closing.
+     */
+    onHide: Function,
+
+    /**
+     * Position of the SlidingView: 'left', 'right', 'top', 'bottom'.
+     * later).
+     */
+    position: string,
+
+    /**
+     * Whether the menu (of the {@code SideBar}?) is displayed/rendered/shown.
+     */
+    show: boolean,
+
+    /**
+     * Style of the animated view.
+     */
+    style: StyleType
+};
+
+/**
+ * The type of the React {@code Component} state of {@link SideBar}.
+ */
+type State = {
+
+    /**
+     * Whether the sliding overlay should be displayed/rendered/shown.
+     */
+    showOverlay: boolean,
+
+    /**
+     * The native animation object.
+     */
+    sliderAnimation: Animated.Value,
```

Comment:
not sure why flow doesn't complain (sigh) but this should have a typeof I think

Reply:
It's probably an exported type or Class, so the type definition is valid this way, but will probably be invalid with an added "typeof".

Reply:
👍